tests/inject/karma.conf.js

```diff
@@ -6,22 +9,40 @@ const rollupPluginReplace = require('@rollup/plugin-replace');
 const rollupPluginTypescript = require('@rollup/plugin-typescript');
 const typescript = require('typescript');
 const {getTestDestDir, rootPath} = require('../../tasks/paths');
+const karmaPluginEchoServer = require('./support/echo-server');
 
 /**
- * @param   {import('karma').Config} config
- * @returns {import('karma').ConfigOptions}
+ * @param {LocalConfig} config
+ * @param {Record<string, string>} env
+ * @returns {ConfigOptions}
  */
-function configureKarma(config) {
-    const headless = config.headless || process.env.KARMA_HEADLESS || false;
+function configureKarma(config, env) {
+    const headless = config.headless || env.KARMA_HEADLESS || false;
 
+    /** @type {ConfigOptions} */
     let options = {
+        failOnFailingTestSuite: true,
+        failOnEmptyTestSuite: true,
         basePath: '../..',
         frameworks: ['jasmine'],
         files: [
             'tests/inject/support/customize.ts',
             'tests/inject/support/polyfills.ts',
+            'tests/inject/dynamic/link-override.tests.ts',
             {pattern: 'tests/inject/**/*.tests.ts', watched: false},
         ],
+        plugins: [
+            'karma-chrome-launcher',
+            'karma-coverage',
+            'karma-firefox-launcher',
+            'karma-rollup-preprocessor',
+            'karma-jasmine',
+            'karma-spec-reporter',
+            karmaPluginEchoServer,
```

Comment:
As you're problably aware, Karma has been implicitly loading all Karma plugins by just loading `./node_modules/karma-*/`. 

Unfortunately this is not possible for the echo server plugin unless I set it up as an actual package, which I decided was overkill. But if I add `karmaPluginEchoServer` to the list, Karma complains until I add everything to the list.